timemachine/optimize/protocol.py

```diff
@@ -47,16 +51,22 @@
 from jax import Array, jit
 from jax import numpy as jnp
 from jax import vmap
-from jax.scipy.special import logsumexp
+from jax.scipy.special import logsumexp as _logsumexp
 from jax.typing import ArrayLike
 from scipy.optimize import bisect
 
+from timemachine.fe.reweighting import interpret_as_mixture_potential
+
+# performance bottleneck in several functions below
+logsumexp = jit(_logsumexp, static_argnames=["axis"])
```

Comment:
nit: would it be preferable to instead add a `@jax.jit` decorator to `reweighted_pair_overlap`? (This looked like the only usage in this file, but might have missed something.)

Reply:
This would be more natural and provide additional opportunities for optimization, but I had hit a snag when I first attempted this. (I did not immediately see why (and haven't investigated), but jit compilation appeared to hang / take an excessive amount of time when `jit` is invoked on the function `reweighted_pair_overlap`. )

And a couple other functions in this file, such as `log_weights_from_mixture` and `reweighted_stddev`, also use this function.